packages/fuel-gauge/src/contract.test.ts

```diff
@@ -1111,4 +1116,61 @@ describe('Contract', () => {
     expect(scriptGasLimit?.toNumber()).toBe(gasLimit);
     expect(bn(maxFeePolicy?.data).toNumber()).toBe(maxFee);
   });
+
+  it('can proper set a proxy contract and use it to call a target contract', async () => {
+    using launch = await launchTestNode({
+      contractsConfigs: [{ factory: CoverageContractFactory }],
+    });
+
+    const {
+      wallets: [wallet],
+      contracts: [coverageContract],
+    } = launch;
+
+    const proxyFactory = await ProxySrc14ContractFactory.deploy(wallet, {
+      storageSlots: ProxySrc14Contract.storageSlots,
+    });
+
+    const { contract: proxyContract } = await proxyFactory.waitForResult();
+
+    // Setting the contract target at the proxy
+    const call = await proxyContract.functions
+      .set_proxy_target({ bits: coverageContract.id.toB256() })
+      .call();
+
+    await call.waitForResult();
+
+    const proxyId = proxyContract.id;
+    const echoValuesId = coverageContract.id;
+
+    const proxiedConverageContract = new Contract(
+      echoValuesId,
+      CoverageContract.abi,
+      wallet,
+      proxyId
```

Comment:
Do you think we could just read this value from the SRC14 storage? Then we could just pass some `{ proxy: true }` config and the user only needs to be aware of one contract ID that doesn't change. 

Only issue is in the contract call, we'd first need to read the value, with a function call to `proxy_target` using the existing contract ID. Then, we'd perform the user dictated function call against the received contract ID.

Reply:
We could then possibly use the `proxy` flag to modify the `Interface` and merge the passed ABI with an `SRC14` ABI?

Reply:
@danielbate I was planning to solve this with Typegen. Since the flag `[proxy]` is available in the `Forc` file, then we could use it. So when connecting with the contract or deploying it, if the flag is present we can already handle things on the fly.

With that, I believe we solve the problem of having the user manually inform the proxy contract ID and avoid executing the call to read the target contract at the proxy contract, which would represent an additional `dryRun` since it is a read call.

About modifying the proxy contract `Interface`, I made a couple of tries but didn't succeed. Maybe I am missing something.

Reply:
Typegen is a good idea, will we still need a way for users to connect that haven't used a typegen factory though? Just a standard `new Contract()`?

Even with typegen though, how do we know the ID of the target contract without the `dryRun`, if it hasn't been deployed by us?

> About modifying the proxy contract Interface, I made a couple of tries but didn't succeed. Maybe I am missing something.

Lets see if we can solve this later, not a massive deal as the user will ideally just want to call the target contract functions anyways.

Reply:
> Even with typegen though, how do we know the ID of the target contract without the dryRun, if it hasn't been deployed by us?

@danielbate It seems the solution you idealize for this is different from the one that I am approaching. In the context described here, the user would have only in its hands the contract Proxy ID and wants to execute a contract call for the target proxy, and they do not have the ID of the deployed target contract, right?

This is why you are suggesting a way to modify the proxy contract interface using the target contract interface. Therefore the user would use just the proxy contract ID to execute the call, then, behind the scenes, we would:

1 - Instantiate the Proxy contract, using also the target contract JSON ABI to have all its existent functions.

2 - Execute a read only call to the Proxy contract to get the target contract ID

3 - Build the script for the contract with the Proxy contract ID and execute the actual contract call specified by the user.

Is that your proposal?

Reply:
Yes exactly, well summarised.